File: rb/features.rb
# Insult SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module InsultFeatures
  def self.make_feature(name)
    case name
    when "base"
      InsultBaseFeature.new
    when "test"
      InsultTestFeature.new
    else
      InsultBaseFeature.new
    end
  end
end

File: py/features.py
# Insult SDK feature factory

from feature.base_feature import InsultBaseFeature
from feature.test_feature import InsultTestFeature


def _make_feature(name):
    features = {
        "base": lambda: InsultBaseFeature(),
        "test": lambda: InsultTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
